api/src/main/java/com/datatorrent/api/Context.java

```diff
@@ -335,10 +335,16 @@
     Attribute<String> APPLICATION_DATA_LINK = new Attribute<String>(new String2String());
     /**
      * Transport to push the stats and the metrics, "builtin:{topic}" if STRAM should push the data directly
-     * using websocket with the given topic
+     * using websocket with the given topic.
+     * If using a custom transport, please use Object2String codec to specify the transport object.  The object must
+     * be from a class that implements the AutoMetric.Transport interface
      */
     Attribute<String> METRICS_TRANSPORT = new Attribute<String>(new String2String());
```

Comment:
As discussed, maybe look at the other option again to use a placeholder object for the "built-in" transport and swap it out for the engine internal implementation when wiring the dependencies in the AM?